### ISSUE TITLE
Duplicate class errors (without classes and sources both present)

### ISSUE THREAD

**Issue by orionlee** [georgewfraser/vscode-javac#34]
I kept getting, Duplicate class errors.  similar to #14 but I believe it is different, 

In the gradle-based (Android) project, It happens to most of the classes, except the class in the initial open editor window. E.g., if `SomeClass.java` is the initial open editor window upon startup, it parses okay. However, if the same `SomeClass.java` is not the initial open editor window, when I subsequently open it, I get Duplicate class errors.

Unlike the scenario in #14, the `"outputDirectory"` does not point to any generated classes either. (It points to `build/vscode-classes`, which is empty, and not being used by the gradle build).

`javac-services.0.log` did not produce any helpful messages either, only stating it is linting the file:

```
INFO    ?   main    2016-10-08T17:27:01.179Z    org.javacs.JavaLanguageServer#doLint    Lint AntennaPod\core\src\main\java\de\danoeh\antennapod\core\service\playback\PlaybackService.java
```

vscode developer tools panel did not give any messages either.

Thoughts?

**Comment by georgewfraser** [comment]
If you could create a minimal reproducible error, ideally *without* gradle, I would be happy to take a look at it.

**Comment by orionlee** [comment]
@georgewfraser I am afraid I don't have enough time to isolate the error further in near term. If there is some settings that I can tweak to produce additional logging info to help investigation that I can do quickly, please let me know.

**Comment by georgewfraser** [comment]
I did some overall simplification over the weekend that might have improved this situation - could you update the plugin and see if you're still getting the issue?

**Comment by orionlee** [comment]
I tried the latest (v0.0.23) and still ran into the issue.

I could clarify the condition the duplicate class error happens for me:
1. The case I've tested have two sub-projects, `app` and `core` [*]
2. The class in the initial opened editor window, e.g, `Foo1.class` in `app` project, works fine.
3. If I switch to (or open) another window of the same project, I get duplicate class error.
4. If I switch to (or open) another window of 2nd project, e.g., `Bar2.class` in `core` project, it works fine.
5. Any subsequent window of the same project will still show class error.

Another way to describe the behavior, going by what shows up in `javac-services.0.log`: 
Upon linting a file, i.e., `main ... org.javacs.JavaLanguageServer#doLint`, 
    1. if the file has been previously indexed, aka `InitialIndex ... org.javacs.SymbolIndex$1#parseAll` , it will show duplicate class error in vscode UI (no such indicator in the log file)
    2. if the file has not been previously indexed, it works fine.

The project in question is an Android project, the issue at hand does not seem to be specific to Android though. I mention it just in case.
